sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala

```diff
@@ -165,4 +154,222 @@ class ShowTablesSuite extends ShowTablesSuiteBase with CommandSuiteBase {
+          s"$defaultUsing PARTITIONED BY (id1)")
+        sql(s"CREATE TABLE $catalog.$namespace.$table2 (id2 bigint, data2 string) " +
+          s"$defaultUsing PARTITIONED BY (id2)")
+
+        val result = sql(s"SHOW TABLE EXTENDED FROM $catalog.$namespace LIKE '$table*'")
+          .sort("tableName")
+        assert(result.schema.fieldNames ===
+          Seq("namespace", "tableName", "isTemporary", "information"))
+
+        val resultCollect = result.collect()
+        assert(resultCollect.length == 3)
+
+        assert(resultCollect(0).length == 4)
+        assert(resultCollect(0)(1) === table)
+        assert(resultCollect(0)(2) === false)
+        val actualResult_0_3 = exclude(resultCollect(0)(3).toString)
+
+        // exclude "Created Time", "Last Access", "Created By", "Location"
+        val expectedResult_0_3 =
+          s"""Catalog: $catalog
+             |Database: $namespace
+             |Table: $table
+             |Type: MANAGED
+             |Provider: parquet
+             |Partition Provider: Catalog
+             |Partition Columns: [`id`]
+             |Schema: root
+             | |-- data: string (nullable = true)
+             | |-- id: long (nullable = true)""".stripMargin
+
+        assert(actualResult_0_3 === expectedResult_0_3)
+
+        assert(resultCollect(1).length == 4)
+        assert(resultCollect(1)(1) === table1)
+        assert(resultCollect(1)(2) === false)
+        val actualResult_1_3 = exclude(resultCollect(1)(3).toString)
+
+        // exclude "Created Time", "Last Access", "Created By", "Location"
+        val expectedResult_1_3 =
+          s"""Catalog: $catalog
+             |Database: $namespace
+             |Table: $table1
+             |Type: MANAGED
+             |Provider: parquet
+             |Partition Provider: Catalog
+             |Partition Columns: [`id1`]
+             |Schema: root
+             | |-- data1: string (nullable = true)
+             | |-- id1: long (nullable = true)""".stripMargin
+        assert(actualResult_1_3 === expectedResult_1_3)
+
+        assert(resultCollect(2).length == 4)
+        assert(resultCollect(2)(1) === table2)
+        assert(resultCollect(2)(2) === false)
+        val actualResult_2_3 = exclude(resultCollect(2)(3).toString)
+
+        // exclude "Created Time", "Last Access", "Created By", "Location"
+        val expectedResult_2_3 =
+          s"""Catalog: $catalog
+             |Database: $namespace
+             |Table: $table2
+             |Type: MANAGED
+             |Provider: parquet
+             |Partition Provider: Catalog
+             |Partition Columns: [`id2`]
+             |Schema: root
+             | |-- data2: string (nullable = true)
+             | |-- id2: long (nullable = true)""".stripMargin
+        assert(actualResult_2_3 === expectedResult_2_3)
+      }
+    }
+  }
+
+  test("show table extended in view: permanent, temp global, temp local") {
```

Comment:
we should have separated tests for temp and permanent views. The v2 command should be able to handle temp views.

Reply:
Done